### PR TITLE
fix(event_list): 11841 - fix autoselect fired when event is already selected

### PR DESCRIPTION
**Pull request by IlyaIzr**
If url was applied with event-id not presented in event list but availible through api call - warning title is presented for 5 seconds and autoselect is not fired 
image

I also added error message for 404 event requests as there was no other way for user to know that event cannot be reached

### CODE REVIEW

**Comment by IlyaIzr** [comment]
i've also changed "event not found in the list" toast color from error to warning state as it's not stopping user from using app properly

File: src/features/events_list/atoms/autoSelectEvent.ts [hunk 1]
```diff
@@ -23,10 +21,20 @@ export const autoSelectEvent = createAtom(
...
+          schedule((dispatch) =>
+            dispatch(
+              currentNotificationAtom.showNotification(
+                'warning',
+                { title: i18n.t('event_list.no_event_in_feed') },
+                5,
+              ),
+            ),
+          );
+        } else if (currentEventNotInNewList) {
```
Comment (propakov) [inline src/features/events_list/atoms/autoSelectEvent.ts]: i would suggest to get rid of else if logic 
like here (feels simplier)
```
      if (currentEventNotInNewList) {
          if (currentEvent?.id) {
            // This case happens when call for event by provided url id didn't return event
            schedule((dispatch) =>
              dispatch(
                currentNotificationAtom.showNotification(
                  'warning',
                  { title: i18n.t('event_list.no_event_in_feed') },
                  5,
                ),
              ),
            );
          } else {
            const firstEventInList = eventListResource.data[0];
            schedule((dispatch) => {
              dispatch([
                scheduledAutoSelect.setFalse(),
                scheduledAutoFocus.setTrue(),
                currentEventAtom.setCurrentEventId(firstEventInList.eventId),
              ]);
            });
          }
        }

File: src/features/events_list/atoms/autoSelectEvent.ts [hunk 2]
```diff
@@ -23,10 +21,20 @@ export const autoSelectEvent = createAtom(
       ) {
         const currentEvent = getUnlistedState(currentEventAtom);
         const currentEventNotInNewList =
-          eventListResource.data.findIndex(
-            (e) => e.eventId === currentEvent?.id,
-          ) === -1;
-        if (currentEventNotInNewList) {
+          eventListResource.data.findIndex((e) => e.eventId === currentEvent?.id) === -1;
```
Comment (propakov) [inline src/features/events_list/atoms/autoSelectEvent.ts]:
```suggestion
          !eventListResource.data.some((e) => e.eventId === currentEvent?.id);
```

File: src/features/events_list/atoms/autoSelectEvent.ts (hunk 1 above)
Comment (IlyaIzr) [reply src/features/events_list/atoms/autoSelectEvent.ts]: for me it's one more nesting\intendation level

Comment (IlyaIzr) [reply src/features/events_list/atoms/autoSelectEvent.ts]: I did compromise

**Review by propakov — APPROVED** [review]

File: src/features/events_list/atoms/autoSelectEvent.ts [hunk 3]
```diff
@@ -22,11 +20,25 @@ export const autoSelectEvent = createAtom(
...
-          eventListResource.data.findIndex(
-            (e) => e.eventId === currentEvent?.id,
-          ) === -1;
-        if (currentEventNotInNewList) {
+        const currentEventNotInTheList = !eventListResource.data.some(
+          (e) => e.eventId === currentEvent?.id,
+        );
+
+        // do nothing
+        if (!currentEventNotInTheList) return;
```
Comment (propakov) [inline src/features/events_list/atoms/autoSelectEvent.ts]: it's pretty clear why early return is needed :)
Btw, i remember we discussed on FE meetup about return undefined from atom. Is it ok?

Comment (IlyaIzr) [reply src/features/events_list/atoms/autoSelectEvent.ts]: good point, i thought it was stateless atom
i will `return state` then

**Review by Akiyamka — APPROVED** [review]

File: src/features/current_event/atoms/currentEventResource.ts [hunk 4]
```diff
@@ -29,7 +30,13 @@ export const currentEventResourceAtom = createAsyncAtom(
         `/events/${deps.feed.id}/${deps.event.id}`,
         undefined,
         undefined,
-        { signal: abortController.signal, errorsConfig: { dontShowErrors: true } },
+        {
+          signal: abortController.signal,
+          errorsConfig: {
+            dontShowErrors: false,
```
Comment (Akiyamka) [inline src/features/current_event/atoms/currentEventResource.ts:36]: I expect now you will always see error when request canceled, that I guess not what you want

**Pull request merged**